File: java_projects/glue_code/commons-cli/src/main/java/org/apache/commons/cli/BasicParser.java
package org.apache.commons.cli;

import org.graalvm.polyglot.Value;

public class BasicParser extends Parser {
  private static Value clz = ContextInitializer.getPythonClass("/BasicParser.py", "BasicParser");
  private Value obj;

  public BasicParser(Value obj) {
    this.obj = obj;
  }

  public BasicParser() {
    this.obj = clz.newInstance();
  }

  public Value getPythonObject() {
    return obj;
  }

  protected String[] flatten(
      @SuppressWarnings("unused") final Options options,
      final String[] arguments,
      @SuppressWarnings("unused") final boolean stopAtNonOption) {
    //
    // return arguments;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("flatten", options, arguments, stopAtNonOption).as(String[].class);
  }
}

File: data/skeletons/commons-cli/src/main/org/apache/commons/cli/BasicParser.py
# Imports Begin
from src.main.org.apache.commons.cli.Parser import *
from src.main.org.apache.commons.cli.Options import *
import typing

# Imports End


class BasicParser(Parser):

    # Class Fields Begin
    # Class Fields End

    # Class Methods Begin
    def _flatten(
        self, options: Options, arguments: typing.List[str], stopAtNonOption: bool
    ) -> typing.List[str]:
        pass

    # Class Methods End
